web/frontend/src/App.tsx

```diff
@@ -63,32 +75,34 @@ const App: FC = () => {
           <div
             data-testid="content"
             className="app-page mb-auto flex flex-row justify-center items-center w-full">
-            <div>
-              {isLogged ? (
-                <Routes>
-                  <Route path={ROUTE_ELECTION_CREATE} element={<ElectionCreate />} />
-                  <Route path={ROUTE_ELECTION_SHOW + '/:electionId'} element={<ElectionShow />} />
-                  <Route path={ROUTE_RESULT_INDEX} element={<ResultIndex />} />
-                  <Route path={ROUTE_RESULT_SHOW + '/:electionId'} element={<ResultShow />} />
-                  <Route path={ROUTE_BALLOT_INDEX} element={<BallotIndex />} />
-                  <Route path={ROUTE_BALLOT_SHOW + '/:electionId'} element={<BallotShow />} />
-                  <Route path={ROUTE_ADMIN} element={<Admin />} />
-                  <Route path={ROUTE_ABOUT} element={<About />} />
-                  <Route path={ROUTE_ELECTION_INDEX} element={<ElectionIndex />} />
-                  <Route path="/" element={<Home />} />
-                  <Route path="*" element={<NotFound />} />
-                </Routes>
-              ) : (
-                <Routes>
-                  <Route path={ROUTE_ADMIN} element={<Admin />} />
-                  <Route path={ROUTE_ABOUT} element={<About />} />
-                  <Route path={ROUTE_ELECTION_INDEX} element={<ElectionIndex />} />
-                  <Route path="/" element={<Login />} />
-                  <Route path="/login_callback" element={<LoginCallback />} />
-                  <Route path="*" element={<NotFound />} />
-                </Routes>
-              )}
-            </div>
+            <Routes>
+              <Route
+                path={ROUTE_ELECTION_CREATE}
+                element={
+                  <RequireAuth>
+                    <ElectionCreate />
+                  </RequireAuth>
+                }
+              />
+              <Route path={ROUTE_ELECTION_SHOW + '/:electionId'} element={<ElectionShow />} />
+              <Route path={ROUTE_RESULT_INDEX} element={<ResultIndex />} />
+              <Route path={ROUTE_RESULT_SHOW + '/:electionId'} element={<ResultShow />} />
+              <Route path={ROUTE_BALLOT_INDEX} element={<BallotIndex />} />
+              <Route path={ROUTE_BALLOT_SHOW + '/:electionId'} element={<BallotShow />} />
+              <Route
+                path={ROUTE_ADMIN}
+                element={
+                  <RequireAuth>
+                    <Admin />
+                  </RequireAuth>
+                }
+              />
+              <Route path={ROUTE_ABOUT} element={<About />} />
+              <Route path={ROUTE_ELECTION_INDEX} element={<ElectionIndex />} />
+              <Route path={ROUTE_LOGIN} element={<Login />} />
+              <Route path="/" element={<Home />} />
+              <Route path="*" element={<NotFound />} />
```

Comment:
Maybe we should use a dedicated Page404.tsx (under pages/)

Reply:
Yes, that's something to eventually add. Even better would be to have a `Error` component that can render different kind of errors.